datadog_checks_dev/datadog_checks/dev/tooling/commands/validate/imports.py

```diff
@@ -0,0 +1,96 @@
+# (C) Datadog, Inc. 2018-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+import os
+
+import click
+
+from ...constants import get_root
+from ...utils import complete_valid_checks, get_valid_integrations
+from ..console import CONTEXT_SETTINGS, abort, echo_debug, echo_failure, echo_info, echo_success, echo_warning
+
+
+def validate_import(filepath, check, autofix):
+    """Validate imports are coming from the correct base package."""
+    # almost every case is of the form `from datadog_checks.. import ..`
+    # we want to ensure that the imports are from `datadog_checks.base...`
+    # except for cases where its importing from actual check code
+
+    success = True
+    lines = []
+
+    with open(filepath) as f:
+        for num, line in enumerate(f):
+            if all(
+                (
+                    'import' in line,
+                    'datadog_checks' in line,
+                    'datadog_checks.base' not in line,
+                    f'datadog_checks.{check}' not in line,
+                    'datadog_checks.dev' not in line,
+                )
+            ):
+                success = False
+                lines.append((num, line))
+
+    if autofix and not success:
+        with open(filepath, 'r') as f:
+            data = f.readlines()
+
+        for num, _ in lines:
+            data[num] = data[num].replace('datadog_checks', 'datadog_checks.base')
+
+        with open(filepath, 'w') as f:
+            f.write(''.join(data))
+
+    return success, lines
+
+
+@click.command(context_settings=CONTEXT_SETTINGS, short_help='Validate proper base imports')
+@click.argument('checks', nargs=-1, autocompletion=complete_valid_checks, required=False)
+@click.option('--autofix', is_flag=True, help='Apply suggested fix')
+@click.pass_context
+def imports(ctx, autofix, checks):
+    """Validate proper imports in checks."""
+
+    validation_fails = {}
+    integrations_root = get_root()
+    all_checks = sorted(get_valid_integrations())
```

Comment:
I'm not positive, but I'm pretty sure `get_valid_checks() `would suffice, since any non-check integration wouldn't have a python file that imports base
```suggestion
    all_checks = sorted(get_valid_checks())
```

Reply:
In this case, `get_valid_integrations` is a superset, but only really includes a few extra packages: `{'agent_metrics', 'amazon_eks', 'pivotal_pks', 'containerd', 'cri', 'systemd', 'openshift', 'ntp'}`

Shouldn't make too much of an impact since there aren't many python files in those few extra packages, but would have coverage if something gets added there.

Reply:
Yeah I agree it's not necessary to change. But if check code were added to non-check integrations then I think they would be include in the `get_valid_checks()` since it looks for the version file.